plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -227,4 +238,53 @@ private static String quoteIdentifier(String identifier)
     {
         return format("\"%s\"", identifier.replaceAll("\"", "\"\""));
     }
+
+    public static Optional<String> getQueryOptionsString(String options)
+    {
+        if (isNullOrEmpty(options)) {
+            return Optional.empty();
+        }
+
+        Map<String, String> queryOptionsMap = parseQueryOptions(options);
+        return getQueryOptions(queryOptionsMap);
+    }
+
+    public static Optional<String> getQueryOptions(Map<String, String> queryOptionsMap)
+    {
+        if (queryOptionsMap.isEmpty()) {
+            return Optional.empty();
+        }
+        String options = queryOptionsMap.entrySet().stream()
+                .map(e -> "SET " + e.getKey() + " = " + e.getValue())
+                .collect(Collectors.joining(";\n"));
+        if (!options.endsWith(";")) {
+            options += ";";
+        }
+        return Optional.of(options);
+    }
+
+    public static Map<String, String> parseQueryOptions(String options)
+    {
+        if (isNullOrEmpty(options)) {
+            return ImmutableMap.of();
+        }
+        try {
+            // we allow escaping the delimiters like , and : using back-slash.
+            // To support that we create a negative lookbehind of , and : which
+            // are not preceded by a back-slash.
+            String headersDelim = "(?<!\\\\),";
+            String kvDelim = "(?<!\\\\):";
+            Map<String, String> temp = new HashMap<>();
```

Comment:
Use `ImmutableMap.Builder` and rename to `queryOptions`.

Reply:
@ebyhr done